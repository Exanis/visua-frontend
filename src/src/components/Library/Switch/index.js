import React from 'react';
import PropTypes from 'prop-types';

import FormControlLabel from '@material-ui/core/FormControlLabel';
import Switch from '@material-ui/core/Switch';

class VisuaSwitch extends React.Component {
    render () {
        return <FormControlLabel
          control={
            <Switch
              checked={this.props.checked}
              onChange={this.props.onChange}
            />
          }
          label={this.props.label}
        />
    }
}

VisuaSwitch.propTypes = {
    checked: PropTypes.bool.isRequired,
    onChange: PropTypes.func.isRequired,
    label: PropTypes.string.isRequired
};

export default VisuaSwitch;