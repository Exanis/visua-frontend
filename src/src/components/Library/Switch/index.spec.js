import React from 'react';
import { shallow, mount } from 'enzyme';

import Switch from './index';

describe('<Switch />', () => {
    const defaultProps = {
        checked: true,
        onChange: () => {},
        label: 'test-label'
    };

    it('should render Switch properly', () => {
        const rendererComponent = shallow(<Switch {...defaultProps} />);

        expect(rendererComponent).toMatchSnapshot();
    });
});