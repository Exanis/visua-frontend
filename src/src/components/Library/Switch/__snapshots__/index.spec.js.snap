// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> should render Switch properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VisuaSwitch
    checked={true}
    label="test-label"
    onChange={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "control": <WithStyles(Switch)
        checked={true}
        onChange={[Function]}
/>,
      "label": "test-label",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "control": <WithStyles(Switch)
          checked={true}
          onChange={[Function]}
/>,
        "label": "test-label",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
